File: ABC_training/D_diff/python/D_GridColoring.py
#           D - Grid Coloring
# ----------------------------------------
# 問題
# https:#atcoder.jp/contests/abc069/tasks/arc080_b

# 参考
# https://blog.hamayanhamayan.com/entry/2017/08/06/230026

# AC (解説)
# ----------------------------------------

def init_array(i, j, val=0): return [[val]*j for _ in range(i)]
def exprint(x): print(*x, sep="\n")

H, W = map(int, input().split())
N = int(input())
A = list(map(int, input().split()))

res = init_array(W, H)

ptr = 0
for i, a in enumerate(A):
    for _ in range(a):
        res[ptr//H][ptr%H] = i+1
        ptr += 1

# 反転しよう
for i in range(W):
    if i % 2 == 1:
        res[i].reverse()

# 転置しよう
for i in range(H):
    for j in range(W):
        print(res[j][i], end=" ")
    print()

File: abc_training_addition/D_diff/D_GridColoring.cpp
//           D - Grid Coloring
// ----------------------------------------
// 問題
// https://atcoder.jp/contests/abc069/tasks/arc080_b

// PythonでAC
// ----------------------------------------

// 配置の仕方を考える
// 1 6 7
// 2 5 8
// 3 4 9

#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    int H, W, N; cin >> H >> W >> N;
    vector<int> A(N);
    for (auto& a : A) cin >> a;

    int res[H][W];
    int row = 0;
    bool isEven;
    for (int i = 0; i < N; i++) {
        isEven = (i / H) % 2 == 0;  // 偶数行目->左から、奇数行目->右から

        if (isEven) {
            res[row][i/H] = i+1;
            row++;
        } else {
            res[row][i/H] = i+1;
            row--;
        }
    }

    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            cout << res[i][j] << " ";
        }
        cout << endl;
    }
}